File: src/main/java/com/rewards_service/dto/RedeemResponse.java
package com.rewards_service.dto;

import java.math.BigDecimal;
import java.util.UUID;

public class RedeemResponse {

    private UUID userId;
    private long pointsRedeemed;
    private BigDecimal monetaryValue;
    private UUID transactionId;

    public RedeemResponse(UUID userId, long pointsRedeemed, BigDecimal monetaryValue, UUID transactionId) {
        this.userId = userId;
        this.pointsRedeemed = pointsRedeemed;
        this.monetaryValue = monetaryValue;
        this.transactionId = transactionId;
    }

    // Getters
    public UUID getUserId() {
        return userId;
    }

    public long getPointsRedeemed() {
        return pointsRedeemed;
    }

    public BigDecimal getMonetaryValue() {
        return monetaryValue;
    }

    public UUID getTransactionId() {
        return transactionId;
    }
}

File: src/main/kotlin/com/rewards_service/dto/RedeemResponse.kt
package com.rewards_service.dto

import java.math.BigDecimal
import java.util.UUID

class RedeemResponse {

    var userId: UUID? = null
    var pointsRedeemed: Long = 0L
    var monetaryValue: BigDecimal? = null
    var transactionId: UUID? = null

    constructor(userId: UUID?, pointsRedeemed: Long, monetaryValue: BigDecimal?, transactionId: UUID?) {
        this.userId = userId
        this.pointsRedeemed = pointsRedeemed
        this.monetaryValue = monetaryValue
        this.transactionId = transactionId
    }
}
